### PR TITLE
Add tests for the element section

### PR DESCRIPTION
**Pull request by gahaas**
Some of the tests were moved from call_indirect.wast.

### CODE REVIEW

File: test/core/elements.wast [hunk 1]
```diff
@@ -0,0 +1,163 @@
+;; Test the element section
```
Comment (rossberg-chromium) [inline test/core/elements.wast]: Nit: most tests follow the convention to be named after the keyword of the construct they test. Therefore, can we name the file `elem`?

**Comment by gahaas** [comment]
I renamed the test file now, and added a test where element sections across multiple modules initialize the same table.

File: test/core/elem.wast [hunk 2]
```diff
@@ -161,3 +161,54 @@
...
+  (func (export "call-8") (type $out-i32)
+    (call_indirect $out-i32 (i32.const 8))
+  )
+  (func (export "call-9") (type $out-i32)
+    (call_indirect $out-i32 (i32.const 9))
+  )
+  (export "shared-table" (table 0))
+)
+
+(register "test")
```
Comment (rossberg-chromium) [inline test/core/elem.wast]: Nit: `(register "test" $module-1)` for extra clarity. Also, maybe give it a more descriptive name than "test", or call it "module-1"..

File: test/core/elem.wast [hunk 3]
```diff
@@ -161,3 +161,54 @@
...
+  (func (export "call-7") (type $out-i32)
+    (call_indirect $out-i32 (i32.const 7))
+  )
+  (func (export "call-8") (type $out-i32)
+    (call_indirect $out-i32 (i32.const 8))
+  )
+  (func (export "call-9") (type $out-i32)
+    (call_indirect $out-i32 (i32.const 9))
+  )
+  (export "shared-table" (table 0))
```
Comment (rossberg-chromium) [inline test/core/elem.wast]: Nit: you can merge this into the table definition, like for functions.

File: test/core/elem.wast [hunk 4]
```diff
@@ -161,3 +161,54 @@
...
+)
+
+(register "test")
+
+(assert_return (invoke "call-8") (i32.const 65))
```
Comment (rossberg-chromium) [inline test/core/elem.wast]: Can you also add `(assert_trap (invoke "call-7") "...")`?

File: test/core/elem.wast (hunk 3 above)
Comment (gahaas) [inline test/core/elem.wast]: Done.

File: test/core/elem.wast (hunk 2 above)
Comment (gahaas) [inline test/core/elem.wast]: Done.

File: test/core/elem.wast (hunk 4 above)
Comment (gahaas) [inline test/core/elem.wast]: Done.

**Pull request merged**